SocialCareCaseViewerApi/V1/Boundary/Requests/AddNewResidentRequest.cs

```diff
@@ -61,5 +61,9 @@ public string EmailAddress
         [Required]
         [EmailAddress]
         public string CreatedBy { get; set; }
+
+        [MaxLength(1)]
+        [RegularExpression("(?i:^Y|N)", ErrorMessage = "The restricted must be 'Y' or 'N' only.")]
+        public string Restricted { get; set; } = "N";
```

Comment:
How does this validation work? Do we have to catch an exception in the controller for the endpoint to return a 400 bad request? Just wondering if we have to do something similar like creating allocations:

https://github.com/LBHackney-IT/social-care-case-viewer-api/blob/f2e3e4943e153f402ad890f9470b722b346d123d/SocialCareCaseViewerApi/V1/Controllers/SocialCareCaseViewerApiController.cs#L229-L237

Reply:
This is using out of the box .NET validation attributes, so the validation and error messages are handled automatically by the framework based on the model. Allocation is using Fluent validation library now which requires some additional setup. We are slowly refactoring stuff to use that one, but the person and most other end points are still using the default approach.

Reply:
Ahh cooool, that makes sense. Thanks for the explanation!